bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/handler/JuiceNetDeviceHandler.java

```diff
@@ -0,0 +1,330 @@
+                        case "stop":
+                            if (targetTimeTou == 0) {
+                                targetTimeTou = deviceStatus.target_time;
+                            }
+                            overrideTime = deviceStatus.unit_time + 31556926;
+                            break;
+                        case "start":
+                            if (targetTimeTou == 0) {
+                                targetTimeTou = deviceStatus.target_time;
+                            }
+                            overrideTime = deviceStatus.unit_time;
+                            break;
+                        case "smart":
+                            overrideTime = deviceStatus.default_target_time;
+                            break;
+                    }
+
+                    getApi().setOverride(token, energyAtPlugin, overrideTime, energyToAdd);
+
+                    break;
+                }
+            }
+        } catch (JuiceNetApiException | IOException | InterruptedException | TimeoutException | ExecutionException e) {
+            handleApiException(e);
+            return;
+        }
+    }
+
+    public void _queryDeviceStatusAndInfo()
+            throws JuiceNetApiException, IOException, InterruptedException, TimeoutException, ExecutionException {
+        deviceStatus = getApi().queryDeviceStatus(token);
+
+        if (deviceStatus.info_timestamp > lastInfoTimestamp) {
+            lastInfoTimestamp = deviceStatus.info_timestamp;
+
+            deviceInfo = getApi().queryInfo(token);
+            deviceTouSchedule = getApi().queryTOUSchedule(token);
+            refreshInfoChannels();
+        }
+
+        int carId = deviceStatus.car_id;
+        for (JuiceNetApi.JuiceNetApiCar car : deviceInfo.cars) {
+            if (car.car_id == carId) {
+                this.deviceCar = car;
+                break;
+            }
+        }
+
+        refreshStatusChannels();
+    }
+
+    public void queryDeviceStatusAndInfo() {
+        if (this.getThing().getStatus() == ThingStatus.OFFLINE) {
+            goOnline();
+            // _queryDeviceStatusAndInfo is called by goOnline
+            return;
+        }
+
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            return;
+        }
+
+        try {
+            _queryDeviceStatusAndInfo();
+        } catch (JuiceNetApiException | IOException | InterruptedException | TimeoutException | ExecutionException e) {
+            handleApiException(e);
+            return;
+        }
+    }
+
+    protected ZonedDateTime toZonedDateTime(long localEpochSeconds) {
+        Instant instant = Instant.ofEpochSecond(localEpochSeconds);
+        ZonedDateTime zdt = instant.atZone(ZoneId.of("UTC"));
+
+        Bridge bridge = Objects.requireNonNull(getBridge());
+        JuiceNetBridgeHandler handler = (JuiceNetBridgeHandler) Objects.requireNonNull(bridge.getHandler());
+
+        return zdt.withZoneSameLocal(handler.getTimeZoneProvider().getTimeZone());
```

Comment:
This could be simplified by having `TimeZoneProvider` stored in the handler instance:
```suggestion
        return Instant.ofEpochSecond(localEpochSeconds).atZone(timeZoneProvider.getTimeZone());
```

Reply:
I originally had it implemented that way, but then thought it would be better to store only once in the bridge vs. each device.  I agree though, this is simpler.

Reply:
Leaving this comment open, since only half of the proposal was adopted. :-) You do not need the temporary `ZonedDateTime` in UTC.